sti-kbsearch/src/main/java/uk/ac/shef/dcs/kbsearch/KBSearchFactory.java

```diff
@@ -1,58 +1,61 @@
 import uk.ac.shef.dcs.kbsearch.sparql.DBpediaSearch;
 
 import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 import java.util.Properties;
 
 /**
  * Created by - on 17/03/2016.
  */
 public class KBSearchFactory {
+  private static final String PROPERTIES_SEPARATOR = "\\|";
 
+  public Collection<KBSearch> createInstance(
+          String kbSearchPropertyFiles,
+          String cachesBasePath) throws KBSearchException {
 
-    public KBSearch createInstance(
-            String kbSearchPropertyFile,
-            EmbeddedSolrServer cacheEntity, EmbeddedSolrServer cacheConcept,
-            EmbeddedSolrServer cacheProperty,EmbeddedSolrServer cacheSimilarity) throws KBSearchException {
-
+    try {
+      List<KBSearch> result = new ArrayList<>();
+      String[] kbSearchPropertyFilesArray = kbSearchPropertyFiles.split(PROPERTIES_SEPARATOR);
+      for (String kbSearchPropertyFile : kbSearchPropertyFilesArray) {
         Properties properties = new Properties();
-        try {
-            properties.load(new FileInputStream(kbSearchPropertyFile));
-
-            String className = properties.getProperty(KBSearch.KB_SEARCH_CLASS);
-            boolean fuzzyKeywords = Boolean.valueOf(properties.getProperty(KBSearch.KB_SEARCH_TRY_FUZZY_KEYWORD,"false"));
-            if (className.equals(FreebaseSearch.class.getName())) {
-                return (KBSearch) Class.forName(className).
-                        getDeclaredConstructor(Properties.class,
-                                Boolean.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class).
-                        newInstance(properties,
-                                fuzzyKeywords, cacheEntity, cacheConcept, cacheProperty,
-                                cacheSimilarity);
-            }else if(className.equals(DBpediaSearch.class.getName())){
-                return (KBSearch) Class.forName(className).
-                        getDeclaredConstructor(Properties.class,
-                                Boolean.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class,
-                                EmbeddedSolrServer.class).
-                        newInstance(properties,
-                                fuzzyKeywords, cacheEntity, cacheConcept, cacheProperty,
-                                cacheSimilarity);
-            }
-
-            else {
-                throw new KBSearchException("Class:" + className + " not supported");
-            }
-        } catch (Exception e) {
-            throw new KBSearchException(e);
+        properties.load(new FileInputStream(kbSearchPropertyFile));
```

Comment:
It works for me, because my Java working directory is same as working directory in the main properties file. It might be best to set the working directory for Java to the value set in the main properties file.